File: bun_hono/src/services/userService.ts
import type { AppState } from '../config/state';
import type { UserResponse } from '../models/auth';

export const listUsers = async (state: AppState): Promise<UserResponse[]> => {
  const users = await state.db<UserResponse[]>`
    SELECT id, email FROM users ORDER BY email ASC
  `;
  return users;
};

File: back-end/src/services/user_service.rs
use crate::{error::AppError, models::auth::UserResponse, state::AppState};

pub async fn list_users(state: &AppState) -> Result<Vec<UserResponse>, AppError> {
    let users =
        sqlx::query_as::<_, UserResponse>("SELECT id, email, role FROM users ORDER BY email ASC")
            .fetch_all(&state.db)
            .await?;

    Ok(users)
}
